nodejs/eks/authenticationMode.test.ts

```diff
@@ -70,10 +70,20 @@ describe("validateAuthenticationMode", () => {
         };
 
         expect(() => validateAuthenticationMode(args)).toThrowError(
-            "The 'instanceRoles' property is not supported when 'authenticationMode' is set to 'API'.",
+            "The 'instanceRoles' property does not support non-empty values when 'authenticationMode' is set to 'API'.",
```

Comment:
Actually something must be of as this test shouldn't compile.

```
 const args = {
            authenticationMode: "API",
            instanceRoles: ["roleArn"],
        };
```

"roleArn" is not a valid aws.eks.Role.

Reply:
The test isn't type checked it's executed as js. If you'd like to satisfy the type checker you could pass the `args as ClusterOptions`

Reply:
I've added it to tsconfig.json and fixed it up, it now is type-checked.